### DISCUSSION TITLE
Documentation about mocks unclear

### DISCUSSION THREAD

**Discussion by smares** [vitest-dev/vitest#1370] [Q&A]
https://vitest.dev/guide/mocking.html#modules states

> If your code is importing mocked module, without any associated __mocks__ file or factory for this module, Vitest will mock the module itself by invoking it and mocking every export.

I assumed this means doing `vi.mock('axios')` will be enough to have a mock of axios with all its methods being `vi.fn()`s, but in fact, all of the methods I assumed would be there are undefined.

To demonstrate this, I put together a quick demo on StackBlitz which in a second test returns a factory for `vi.mock('axios')` that mocks the `get` function, but this also doesn't work on StackBlitz while it works in my environment; weird: https://stackblitz.com/edit/vitest-dev-vitest-vtxq7v?file=test/axios.test.ts

1. Any idea why my second test fails on StackBlitz?
2. Is there a bug with mocking or did I just misunderstand the documentation?

**Comment by sheremet-va** [comment]
1. All `vi.mock` calls are hoisted to the top of the file, so only one of the `vi.mock` matters.
2. It seems automocking detected that `default` is a function and created a spy function for it without putting other properties on it. Seems like a bug, I guess

**Reply by smares** [reply]
Thanks for the quick reply!

**Reply by smares** [reply]
Is 2. being looked at or should I post an issue? :)

**Reply by Kjir** [reply]
I opened a bug for this issue: #1523

**Comment by Arizzle81** [comment]
Hello - I'm also having issues running the example of the provided [axios mock](https://github.com/vitest-dev/vitest/blob/main/examples/mocks/test/external.test.ts) locally. Would it be possible to show that this is passing in the example please? New to vitest. Thank you.

contents of external.mjs file:
```
import { vi } from 'vitest'
vi.doMock('axios')
```

Example test:
```
import './external.mjs'
import { expect, test, vi } from 'vitest'
import axios from 'axios'

test('axios is mocked', () => {
  expect(vi.isMockFunction(axios.get)).toBe(true)
})
```


```
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  app/audit/example.unit.spec.js > axios is mocked
AssertionError: expected false to be true // Object.is equality
 ❯ app/audit/example.unit.spec.js:6:40
      4| 
      5| test('axios is mocked', () => {
      6|   expect(vi.isMockFunction(axios.get)).toBe(true)
       |                                        ^
      7| })

  - Expected   "true"
  + Received   "false"

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯
```

**Comment by andrewmelnyk** [comment]
I'm also dealing with an axios mock issue although I'm debating if it's a TS configuration issue as well.

I am trying to move away from jest and use vitest entirely in a TS environment but am running into this issue only when I try to use the `.get` method from the axios mocked function.

As you can see casting axios using jest works, but with vitest.Mocked it does not. Am I using this the wrong way? I added a bunch of extra white space so you can see what is typed out.

At the top of my test file I am importing vitest as `import { it, Mocked, vi } from "vitest";`

edit: I actually realized that it works just like jest after doing what is now obvious

import {Mock} from 'vitest'
// vitest
(axios.post as Mock)
// jest
(axios.post as jest.Mock)
